### PR TITLE
feat: create indexing to improved query performance

### PR DESCRIPTION
**Pull request by nwjbrandon**

### CODE REVIEW

**Comment by Merrickneo** [comment]
I think we can try to maximise the use of indexes. So each table could minimally one index based on the criteria that we do the filtering on (field in the WHERE clause).

File: setup/distributed/schema_v4.sql [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+	D_STATE CHAR(2),
+	D_ZIP CHAR(9),
+	D_TAX DECIMAL(4, 4),
+	D_YTD DECIMAL(12, 2),
+	D_NEXT_O_ID INT,
+	PRIMARY KEY (D_W_ID, D_ID),
+	FOREIGN KEY (D_W_ID) REFERENCES Warehouse(W_ID)
+);
+SELECT create_distributed_table('district', 'd_w_id', colocate_with => 'warehouse');
+
```
Comment (Merrickneo) [inline setup/distributed/schema_v4.sql:35]: `CREATE INDEX "district_idx" ON "district" (D_ID, D_W_ID);`

Comment (Merrickneo) [reply setup/distributed/schema_v4.sql:35]: Can do the same for all vertically partitioned tables for `District`

File: setup/distributed/schema_v4.sql [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+CREATE TABLE item (
+    I_ID INT PRIMARY KEY,
+    I_NAME VARCHAR(24),
+    I_PRICE DECIMAL(5, 2),
+    I_IM_ID INT,
+    I_DATA VARCHAR(50)
+);
+SELECT create_distributed_table('item', 'i_id');
+
```
Comment (Merrickneo) [inline setup/distributed/schema_v4.sql:128]: Add an index based on the `I_ID` field

**Comment by nwjbrandon** [comment]
> I think we can try to maximise the use of indexes. So each table could minimally one index based on the criteria that we do the filtering on (field in the WHERE clause).

I have added indexes for every table to optimize the where clause.

**Pull request merged**